aiomisc/compat.py

```diff
@@ -0,0 +1,105 @@
+import asyncio
+import logging
+import socket
+from functools import partial
+from typing import Any, Callable, Optional, TypeVar
+
+
+log = logging.getLogger(__name__)
+
+
+try:
+    # noinspection PyUnresolvedReferences
+    import contextvars
+
+    T = TypeVar("T")
+    F = TypeVar("F", bound=Callable[..., Any])
+
+    def context_partial(
+        func: F, *args: Any,
+        **kwargs: Any
+    ) -> Any:
+        context = contextvars.copy_context()
+        return partial(context.run, func, *args, **kwargs)
+except ImportError:
+    context_partial = partial   # type: ignore
+
+
+try:
+    from time import time_ns
+except ImportError:
+    from time import time
+
+    def time_ns() -> int:
+        return int(time() * 1000000000)
+
+
+try:
+    from queue import SimpleQueue
+except ImportError:
+    from queue import Queue as SimpleQueue  # type: ignore
+
+
+def _get_running_loop_backport() -> asyncio.AbstractEventLoop:
+    loop = asyncio.get_event_loop()
+    if loop.is_running():
+        return loop
+    raise RuntimeError("no running event loop")
+
+
+get_running_loop = getattr(
+    asyncio, "get_running_loop", _get_running_loop_backport,
+)
```

Comment:
Maybe we should drop python3.6 support? It already reached end-of-life.

Reply:
it's unfortunately most popular python in 2021